qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java

```diff
@@ -8,202 +8,154 @@
+@Testcontainers
 final class ControlledActorClockEndpointIT {
-  private static final String ELASTICSEARCH_HOST = "elasticsearch";
-  private static final String INDEX_PREFIX = "exporter-clock-test";
-  private Network network;
-  private ElasticsearchContainer elasticsearchContainer;
-  private ZeebeContainer zeebeContainer;
-  private ZeebeClient zeebeClient;
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper().registerModule(new JavaTimeModule());
+
+  private final ZeebeContainer container =
+      new ZeebeContainer(ZeebeTestContainerDefaults.defaultTestImage())
+          .withEnv("ZEEBE_CLOCK_CONTROLLED", "true");
+
+  @Container
+  private final ContainerEngine engine =
+      ContainerEngine.builder()
+          .withContainer(container)
+          .withIdlePeriod(Duration.ofSeconds(2))
+          .build();
+
   private final HttpClient httpClient = HttpClient.newHttpClient();
-  private final ObjectMapper mapper = new ObjectMapper().registerModule(new ZeebeProtocolModule());
-  private ZeebeClock zeebeClock;
+  private ZeebeClient zeebeClient;
 
   @BeforeEach
-  void startContainers() {
-    network = Network.newNetwork();
-    startElasticsearch();
-    startZeebe();
-  }
-
-  @AfterEach
-  void stopContainers() {
-    CloseHelper.closeAll(zeebeClient, zeebeContainer, elasticsearchContainer, network);
+  void beforeEach() {
+    zeebeClient = engine.createClient();
   }
 
   @Test
-  void testPinningTime() throws IOException, InterruptedException {
+  void testPinningTime() throws IOException, InterruptedException, TimeoutException {
     // given - Zeebe actor clock is pinned
-    final var pinnedAt = zeebeClock.pinTime(Instant.now().minus(Duration.ofDays(3)));
+    final var now = System.currentTimeMillis();
+    final var request =
+        buildRequest("pin")
+            .POST(BodyPublishers.ofByteArray(MAPPER.writeValueAsBytes(Map.of("epochMilli", now))))
+            .build();
     final var process = Bpmn.createExecutableProcess().startEvent().endEvent().done();
+    final var response = httpClient.send(request, new ResponseHandler());
 
     // when - producing records
     zeebeClient.newDeployResourceCommand().addProcessModel(process, "process.bpmn").send().join();
 
     // then - records are exported with a timestamp matching the pinned time
-    waitForExportedRecords();
-    final var records = searchExportedRecords();
-    assertThat(records)
-        .isNotEmpty() // double check that we have exported some records
-        .allSatisfy(record -> assertThat(record.getTimestamp()).isEqualTo(pinnedAt.toEpochMilli()));
+    engine.waitForIdleState(Duration.ofSeconds(5));
+    assertThat(response.statusCode()).as("/pin request was successful").isEqualTo(200);
+    assertThat(response.body().epochMilli()).as("response returned pinned time").isEqualTo(now);
+    assertThat(BpmnAssert.getRecordStream().deploymentRecords())
```

Comment:
:thought_balloon: It would have been nice if we had some more generic extensions, but I understand `BpmnAssert` is very much user oriented, so that's OK. I think we might see it shine more in the update-tests, for example.

Reply:
It wasn't really built for this purpose indeed. But this why we kept the record stream accessible, allowing users to easily extend with their own assertions if need be.